ethers/provider.nim

```diff
@@ -38,6 +41,9 @@ type
   TransactionResponse* = object
     provider*: Provider
     hash* {.serialize.}: TransactionHash
+    convertCustomErrors*: ConvertCustomErrors
```

Comment:
I agree, the callback in `TransactionResponse` is clunky. I tried your suggestion of introducing different return types, but couldn't get it to work.

I first tried to gensym the `Confirmable` type as you suggested. This means that in our `contract` macro we need to change the return type of the generated proc from `Confirmable` to `Confirmable'gensym`, but that leads to a compiler error in chronos. It complains that the return type is changed.

I then tried to leave the return type alone, and work with a type that inherits from `Confirmable`, and use dynamic dispatch (methods) to have different implementations of `confirm()` for error handling. But this didn't work, because `method`s can only be defined at the top level of a file, whereas we allow our `{.contract.}` procs to be defined anywhere.

So I ended up keeping the callback. I did move it into the `Confirmable` type (which is now no longer a type alias, but it's own object type), which ensures that the provider and the TransactionResponse no longer need to know about error conversion. Error conversions is now localized to the `contract.nim` module.